enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/analytics.ts

```diff
@@ -0,0 +1,205 @@
+    /** @see {@link https://github.com/metabase/metabase/blob/master/enterprise/frontend/src/embedding-sdk-bundle/components/public/dashboard/SdkDashboard.tsx#L127} */
+    with_title: true,
+  },
+  question: {
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute.tsx#L149} */
+    drills: true,
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/embedding-sdk-bundle/components/public/SdkQuestion/SdkQuestion.tsx#L129} */
+    with_downloads: false,
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute.tsx#L145} */
+    with_title: true,
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute.tsx#L157} */
+    is_save_enabled: false,
+  },
+  exploration: {
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute.tsx#L157} */
+    is_save_enabled: false,
+  },
+  browser: {
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/MetabaseBrowser.tsx#L39} */
+    read_only: true,
+  },
+};
+
+const SECOND = 1000;
+
+/**
+ * Because the way it's designed to be consumed when each embed is being registered. There could be multiple embed in a page, so we want to debounce and only call this once.
+ */
+export const debouncedReportAnalytics = _.debounce(
+  reportAnalyticsOnce,
+  1 * SECOND,
+);
+
+/**
+ * This is just a safe-guard to ensure we only call this once for whatever reason.
+ */
+let isAnalyticsReported = false;
+function reportAnalyticsOnce(activeEmbeds: Set<MetabaseEmbedElement>) {
+  if (!isAnalyticsReported) {
+    isAnalyticsReported = true;
+    const [firstEmbed] = activeEmbeds;
+    firstEmbed.sendMessage(
+      "metabase.embed.reportAnalytics",
+      createEmbeddedAnalyticsJsUsage(activeEmbeds),
+    );
+  }
+}
+
+export function createEmbeddedAnalyticsJsUsage(
+  activeEmbeds: Set<MetabaseEmbedElement>,
+): EmbeddedAnalyticsJsEventSchema {
+  const [firstEmbed] = activeEmbeds;
+  let usage = {
+    event: "setup",
+    global: {
+      auth_method: getAuthMethod(firstEmbed),
+    },
+  } as EmbeddedAnalyticsJsEventSchema;
+
+  activeEmbeds.forEach((embed) => {
+    match(embed.properties)
+      .with({ componentName: "metabase-dashboard" }, (properties) => {
+        if (!usage.dashboard) {
+          usage = assocIn(usage, ["dashboard"], {
```

Comment:
funny thing is that modifying this object e.g. `usage.dashboard = {}` will fail on Cypress, so I resort to reassigining values instead.